R/collect.R

```diff
@@ -60,9 +84,217 @@ collect_predictions <- function(x) {
     rlang::abort(msg)
   }
 
-  collector(x, coll_col = coll_col)
+  x <- filter_predictions(x, parameters)
+
+  if (summarize) {
+    x <- average_predictions(x, parameters)
+  } else {
+    x <- collector(x, coll_col = coll_col)
+  }
+  x
+}
+
+filter_predictions <- function(x, parameters) {
+  if (is.null(parameters)) {
+    return(x)
+  }
+  metrics <- attr(x, "metrics")
+  params <- attr(x, "parameters")
+  if (is.null(metrics) | is.null(params)) {
+    rlang::warn(
+      paste(
+        strwrap(
+          paste("The object is missing some attributes; it is probably from",
+                "an earlier version of `tune`. The predictions can't be filtered." ),
+          prefix = ""
+        ),
+        collapse = "\n"
+      )
+    )
+
+    return(x)
+  }
+
+  param_names <- params$id
+  parameters <- dplyr::select(parameters, dplyr::one_of(param_names))
```

Comment:
This could probably be `tidyselect::any_of()`.

Reply:
That's not in the CRAN version right now (right?)

Reply:
It looks like it is; The Google lied!

![image](https://user-images.githubusercontent.com/5731043/74869994-f905f000-5326-11ea-8111-594be146a660.png)